### PR TITLE
Bump @api3/ois to 1.4.0 with necessary zod version bump to 3.20

### PR DESCRIPTION
**Pull request by dcroote**
Bumps `zod` as in https://github.com/api3dao/ois/pull/54 in order to [avoid a zod version mismatch with OIS](https://github.com/api3dao/ois#make-sure-only-one-zod-version-is-installed).

### CODE REVIEW

**Comment by Siegrift** [comment]
Renovate bot should take care of this on Monday.

**Review by Siegrift — APPROVED** [review]
👍 LGTM

**Comment by dcroote** [comment]
Unfortunately it looks like `3.20.0` was a bit of a "big" [minor bump](https://github.com/colinhacks/zod/releases/tag/v3.20) and broke a number of things in validator, which is why CI failed to build here: https://github.com/api3dao/airnode/actions/runs/3692777873/jobs/6252029697

**Comment by dcroote** [comment]
> Renovate bot should take care of this on Monday.

Also I think `zod` is excluded from Renovate: https://github.com/api3dao/airnode/blob/524b47de7fd2d505b6bf357ae5b9e8b5f8ca699e/renovate.json#L7-L8

**Comment by Siegrift** [comment]
> Also I think zod is excluded from Renovate:

Good catch. I forgot we set this because it needed to be in sync with validator.

**Comment by dcroote** [comment]
> Unfortunately it looks like `3.20.0` was a bit of a "big" [minor bump](https://github.com/colinhacks/zod/releases/tag/v3.20) and broke a number of things in validator, which is why CI failed to build here: https://github.com/api3dao/airnode/actions/runs/3692777873/jobs/6252029697

@Siegrift  - I fixed most of the issues originating from this "minor" bump, but it won't build until we release a new OIS version because of [oisSchema](https://github.com/api3dao/airnode/blob/524b47de7fd2d505b6bf357ae5b9e8b5f8ca699e/packages/airnode-validator/src/config/config.ts#L8) and the `zod` version incompatibility issue we [encountered before](https://github.com/api3dao/ois#make-sure-only-one-zod-version-is-installed). Hence, I marked this PR as a draft for the moment.

I recommend publishing a new OIS minor after https://github.com/api3dao/ois/pull/56, following which I'll update this PR an it can be merged. WDYT?

**Comment by dcroote** [comment]
Closes #1598.

File: packages/airnode-validator/src/config/config.ts [hunk 1]
```diff
@@ -122,33 +122,28 @@ export const gasPriceOracleStrategySchema = z.discriminatedUnion('gasPriceStrate
...
-      path: ['gasPriceOracle'],
-    });
-  }
-};
-
 export const gasPriceOracleSchema = z
   .array(gasPriceOracleStrategySchema)
   .nonempty()
-  .superRefine(validateGasPriceOracleStrategies);
+  .superRefine((strategies, ctx) => {
```
Comment (Siegrift) [inline packages/airnode-validator/src/config/config.ts:128]: any particular reason for inlining this?

**Review by Siegrift — APPROVED** [review]
👍 LGTM

**Pull request merged**